source/agora/common/EnrollmentManager.d

```diff
@@ -290,6 +335,48 @@ public class EnrollmentManager
 
         return false;
     }
+
+    /***************************************************************************
+
+        Logs message
+
+        Params:
+            msg = the log message to be logged
+            enroll = the Enrollment object, the information of which will be logged
+
+    ***************************************************************************/
+
+    private void logMessage(string msg, const ref Enrollment enroll) @safe nothrow
+    {
+        try
+        {
+            log.info("{}, enrollment:{}", msg, enroll);
+        }
+        catch (Exception ex)
+        {}
+    }
+
+    /***************************************************************************
+
+        Logs exception message
+
+        Params:
+            msg = the log message to be logged
+            enroll = the Enrollment object, the information of which will be logged
+            ex = the Exception object, the message of which will be logged
+
+    ***************************************************************************/
+
+    private void logException(string msg, const ref Enrollment enroll,
+        const ref Exception ex) @safe nothrow
+    {
+        try
+        {
+            log.info("{}, enrollment:{}, exception:{}", msg, enroll, ex);
```

Comment:
I have a feeling an exception might warrant more than `info` ?
It depends if the exception can happen under normal operating circumstances or not.
For example, we should never log at a relevant log level when reading a stream of data from an *unauthenticated* source because it's a DoS vector.
However, when reading from an authenticated source, we can afford an `info` level.
But if it's an internal error that should not happen, `warn/error/fatal` might be warranted depending on the cases of failure. `info` / `trace` is still possible, if e.g. it's a timeout or something of the like.